### PR TITLE
Add Posts and users

### PR DESCRIPTION
**Pull request by gango-anan**
# In the members-only app project, I:
- Created a Post model and a posts controller.
- Created a resource in the routes file which allows [:new, :create, :index] methods.
- Added #before_actions to restrict access to the #new and #create methods to only signed-in users.
- Added code to the #new method to facilitate the creation of a new post.
- Created a simple form for creating a new post.
- Added code to the #create action, to aid in creating a new post associated with the currently signed-in user
with the ability to redirect to the index view if the post creation was successful.
- Added logic to the posts index view to display the post author's name only if a user is logged in.

### CODE REVIEW

File: app/controllers/posts_controller.rb
Comment (teekaytech) [inline app/controllers/posts_controller.rb]: Kindly remove the commented code.

File: app/controllers/posts_controller.rb
Comment (teekaytech) [inline app/controllers/posts_controller.rb]: It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history

File: app/views/posts/index.html.erb
Comment (teekaytech) [inline app/views/posts/index.html.erb]: Kindly move all view-related logic to the view helper.
There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.
      **Resources**
- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've really done well so far in this project. Your readme is awesome :clap:
However, some changes are required before it can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] You have to create the application routes using `resources :posts, only: [:new, :create, :index]`
- [ ] Files not related to the project should not be committed and added to .gitignore (e.g. .browserslistrc)
- [ ] Check my in-line comments below for other required changes.

**[ OPTIONAL CHANGES ]**

- [ ] Kindly add label to your text fields, most especially the sign-up and login forms.
- [ ] You probably have your reason, but I do not think you need the `user_email` field on the posts model.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by gango-anan** [comment]
@teekaytech, the review was powerful I learned a ton of stuff from it, and I  have Implemented all the required changes including the optional ones, thanks.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General


- [ ] Add route for signout 



### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by gango-anan** [comment]
Hello @kobiyoyo, kindly double-check the issue of the route from your side because in my local environment all is working well and even the live version on Heroku is running well and it's the same version as my local one.

**Review by kevcap — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

### I really like your project! Congrats @gango-anan, you did an excellent job, you have my approval to go ahead and merge this pull request!

## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Pull request merged**
